query-engine/connectors/sql-query-connector/src/model_extensions/scalar_field.rs

```diff
@@ -18,6 +18,21 @@ impl ScalarFieldExt for ScalarField {
             (PrismaValue::Float(f), _) => f.into(),
             (PrismaValue::Boolean(b), _) => b.into(),
             (PrismaValue::DateTime(d), _) => d.with_timezone(&Utc).into(),
+            (PrismaValue::Enum(e), TypeIdentifier::Enum(enum_id)) => {
+                let enum_walker = self.dm.clone().zip(enum_id);
+
+                Value::enum_variant_with_name(e, enum_walker.db_name().to_owned())
+            }
+            (PrismaValue::List(vals), TypeIdentifier::Enum(enum_id)) => {
+                let enum_walker = self.dm.clone().zip(enum_id);
+                let variants: Vec<_> = vals
+                    .into_iter()
+                    .map(|val| val.into_string().unwrap())
+                    .map(EnumVariant::new)
+                    .collect();
+
+                Value::EnumArray(Some(variants), Some(EnumName::new(enum_walker.db_name().to_owned())))
```

Comment:
With https://github.com/prisma/team-orm/issues/410, we are going to add metadata to a QuaintValue, maybe we can have some metadatum to pass the name for instance to the previously existing `Array` variant. 

This is completely unimportant, but maybe I can attempt at refactoring that later?

Reply:
Yeah, this is exactly what I was mentioning to you when discussing the refactor of `enum Value` to `struct Value`.

> have some metadatum to pass the name for instance to the previously existing Array variant.
I'm not entirely sure what you mean by the "previously existing array variant" though! I was mostly thinking of simplifying the `Value::Enum` and `Value::EnumArray` variants by uplifting the `EnumName` in the struct itself:

```rs
enum InnerValue<'a> {
  Enum(Option<EnumVariant<'a>>), // revert back to the original state
  Array(Vec<Value>), // get rid of `EnumArray` and use `Array` again
}

struct Value<'a> {
  inner: InnerValue<'a>,
  enum_name: Option<EnumName<'a>>
}
```

This way we can independently slap the enum name on top of the new `Value` struct, without having to mess with `Value::Enum` and `Value::EnumArray`. It's ugly because now you've got an `enum_name` that's not related to the `Value::Enum` variant directly. And `enum_name` would basically only be `Some` when the variant is either a `Value::Enum` or `Value::Array(vals) if vals.iter().all(|val| val.is_enum())`, but I _think_ that's better than having a separate `Value::EnumAray` variant. Up to debate though!

Reply:
> I'm not entirely sure what you mean by the "previously existing array variant" though! I was mostly thinking of simplifying the Value::Enum and Value::EnumArray variants by uplifting the EnumName in the struct itself. 

Exactly that. The previously existing `Array` variant of the enum, now can model an EnumArray, by having added the metadata to the value. 

> Up to debate though!

Yes, the only downside that I see is that we are adding some metadata, for a very specific enum variant, rather than for various variants, and this 1:1 relationship makes me think that it can also belong to the variant. Still I find it cleanly as metadata.